Tools/_sunrise/changelog/actions_changelogs_since_last_run.py

```
@@ -342,6 +376,12 @@ def get_most_recent_workflow(
         if run["id"] == workflow_run["id"]:
             continue
 
+        if target_id is not None:
+            title = str(run.get("display_title", ""))
+            match = DISCORD_RUN_TITLE_RE.fullmatch(title)
+            if match is None or match.group("target") != target_id:
+                continue
+
         return run
```

Ruff (0.16.1)

[warning] Missing explicit `return` at the end of function able to return non-`None` value

Add explicit `return` statement

(RET503)